pkg/catalog/manager/catalog_sync.go

```diff
@@ -37,6 +37,11 @@ func (m *Manager) Sync(key string, obj *v3.Catalog) (runtime.Object, error) {
 		}
 	}
 
+	// check to see if catalog was processed successfully, but had errors while processing
+	if v3.CatalogConditionProcessed.IsFalse(catalog) && v3.CatalogConditionRefreshed.IsTrue(catalog) {
+		return nil, nil
```

Comment:
I think the comment above can be reworded, saying that it processed successfully but had errors is confusing. I also think there should be a comment explaining why you would return without an error despite process failing.